O -> no_entity, enabled evaluation testsuite

twerkmeister opened the pull request:
**Proposed Changes**
- fixes #814 
- enabled the evaluation testsuite

I noticed in the coverage report of travis that the evaluation.py had 0% coverage, and then figured that the tests need to be moved to the base directory!

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality

tmbo commented:
mhm, ok so this was supposed to go into the `0.11.x` branch (not master). can you please rebase your changes onto that branch, otherwise this will pull all changes into that release that are on master.

twerkmeister commented:
Hm I see, I understand that it would be nice to have the no_entity thing on `0.11.x`. However I also fixed the cross validation eval which was in a quite dire state. And for the smells/bugs I removed I did so with some of the new properties on the training data class, which are only on master.
I think I'll split it up as following
1. for the `no_entity` label I'll create a new branch based off of `0.11.x` and create a new pull request
2. enabling the evaluation test suite and fixing cross validation and its tests i'll do on another branch based off master

tmbo commented:
good old separation of concerns :P
